src/components/toast/Toast.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { ReactNode, useCallback } from 'react';
+
+import { FloatingPortal, useTransitionStyles } from '@floating-ui/react';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
+import { IconDefinition, faCircleInfo, faTriangleExclamation } from '@fortawesome/pro-regular-svg-icons';
+import { faXmark } from '@fortawesome/pro-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { Box } from '../box';
+import { Button, ButtonColorTypes, ButtonProps, ButtonVariants } from '../button';
+import { Icon } from '../icon';
+import { ToastVariant } from './Toast.types';
+import { useToastsContext } from './ToastProvider';
+import { AUTO_DISMISS_TIMEOUT } from './constants';
+
+import { CloseButtonWrapper, Message, ToastIcon, ToastPositionAndLayout } from './Toast.styles';
+
+type ToastContentProps = {
+  i: number;
+  toastId: string;
+  variant: ToastVariant;
+  autoDismiss?: boolean;
+  message: ReactNode;
+  showIcon: boolean;
+  icon?: ReactNode;
```

Comment:
There is a variant of toast without an icon. And default icons depend on the toast variants, so I thought that it would be a good approach. Open to proposals to change it